### DISCUSSION TITLE
Should `getStaticPaths` be preview mode aware?

### DISCUSSION THREAD

**Discussion by wbunting** [vercel/next.js#10995] [Help] [answered]
I could imagine being able to remove paths based on preview mode would be useful for build-time optimization. 

Eg. working on notion-blog https://github.com/ijjk/notion-blog/pull/22 it became clear that you would want to fallback in the case of an unpublished post so that time was not wasted on the production build generating pages for unpublished posts.

**Comment by Timer** [comment, marked as answer]
tl;dr `getStaticPaths` is never re-ran at runtime, including while in preview mode (so it cannot be aware).

By design, `getStaticProps` is meant to handle inputs not returned by `getStaticPaths` (this is one of the reasons `getStaticPaths` cannot pass data to `getStaticProps`).

As there's no fallback behavior in preview mode, running `getStaticPaths` is unnecessary (preview mode treats every path req as-if it were returned from `getStaticPaths`).
